README.md
# DuAn1_Exam

changelog.txt

03:13:26 PM 2019-12-01 --User: binh --action: Sửa câu hỏi --action-id: 15
03:13:49 PM 2019-12-01 --User: binh --action: Xóa đề thi --action-id: DT0003
03:20:37 PM 2019-12-01 --User: anh --action: Thêm tài khoản --action-id: hoa
03:25:06 PM 2019-12-01 --User: hai --action: Làm bài thi --action-id: DT0002
03:27:44 PM 2019-12-01 --User: hoa --action: Làm bài thi --action-id: DT0002
09:55:49 PM 2019-12-01 --User: binh --action: Thêm đề thi --action-id: 
11:51:51 AM 2019-12-02 --User: hai --action: Làm bài thi --action-id: DT0003
02:07:51 PM 2019-12-02 --User: hai --action: Làm bài thi --action-id: DT0003
02:14:59 PM 2019-12-02 --User: hai --action: Làm bài thi --action-id: DT0003
02:48:04 PM 2019-12-02 --User: binh --action: Thêm đề thi --action-id: 
02:48:48 PM 2019-12-02 --User: hai --action: Làm bài thi --action-id: DT0004
02:52:55 PM 2019-12-02 --User: anh --action: Xóa tài khoản --action-id: hoa
02:53:00 PM 2019-12-02 --User: anh --action: Xóa tài khoản --action-id: hoa
02:53:50 PM 2019-12-02 --User: hoa --action: Làm bài thi --action-id: DT0004
02:56:57 PM 2019-12-02 --User: hoa --action: Làm bài thi --action-id: DT0004
10:57:19 AM 2019-12-03 --User: hai --action: Làm bài thi --action-id: DT0003
10:59:56 AM 2019-12-03 --User: hai --action: Làm bài thi --action-id: DT0003
11:14:10 AM 2019-12-03 --User: hai --action: Làm bài thi --action-id: DT0004
09:51:32 AM 2019-12-04 --User: hai --action: Làm bài thi --action-id: DT0003
02:31:29 PM 2019-12-04 --User: binh --action: Thêm đề thi --action-id: 
02:34:21 PM 2019-12-04 --User: hai --action: Làm bài thi --action-id: DT0003
02:35:58 PM 2019-12-04 --User: hai --action: Làm bài thi --action-id: DT0002
02:36:46 PM 2019-12-04 --User: hai --action: Làm bài thi --action-id: DT0004
10:54:08 PM 2019-12-04 --User: hai --action: Làm bài thi --action-id: DT0003
08:18:03 PM 2019-12-05 --User: hai --action: Làm bài thi --action-id: DT0002
08:18:43 PM 2019-12-05 --User: hai --action: Làm bài thi --action-id: DT0004
08:20:46 PM 2019-12-05 --User: hai --action: Làm bài thi --action-id: DT0004
11:15:58 AM 2019-12-07 --User: hai --action: Làm bài thi --action-id: DT0005
11:21:11 AM 2019-12-07 --User: hai --action: Làm bài thi --action-id: DT0002
11:21:27 AM 2019-12-07 --User: hai --action: Làm bài thi --action-id: DT0002
02:19:10 PM 2019-12-09 --User: hai --action: Làm bài thi --action-id: DT0003
02:22:41 PM 2019-12-09 --User: hai --action: Làm bài thi --action-id: DT0002
02:29:08 PM 2019-12-09 --User: hai --action: Làm bài thi --action-id: DT0003
03:29:07 PM 2019-12-09 --User: anh --action: Xóa tài khoản --action-id: hoa
03:29:28 PM 2019-12-09 --User: anh --action: Xóa tài khoản --action-id: hoa
03:34:18 PM 2019-12-09 --User: anh --action: Xóa tài khoản --action-id: hoa
03:38:30 PM 2019-12-09 --User: anh --action: Xóa tài khoản --action-id: hoa
03:45:51 PM 2019-12-10 --User: binh --action: Thêm đề thi --action-id: 
03:46:00 PM 2019-12-10 --User: binh --action: Thêm đề thi --action-id: 
03:46:16 PM 2019-12-10 --User: binh --action: Thêm đề thi --action-id: 
09:38:20 AM 2019-12-11 --User: binh --action: Thêm câu hỏi --action-id: 
09:38:42 AM 2019-12-11 --User: binh --action: Sửa câu hỏi --action-id: 50
09:38:46 AM 2019-12-11 --User: binh --action: Xóa câu hỏi --action-id: 50
09:38:58 AM 2019-12-11 --User: binh --action: Thêm câu hỏi --action-id: 
09:39:40 AM 2019-12-11 --User: binh --action: Thêm câu hỏi --action-id: 
09:40:21 AM 2019-12-11 --User: binh --action: Thêm câu hỏi --action-id: 
09:41:28 AM 2019-12-11 --User: binh --action: Thêm câu hỏi --action-id: 
09:42:13 AM 2019-12-11 --User: binh --action: Thêm câu hỏi --action-id: 
09:42:56 AM 2019-12-11 --User: binh --action: Thêm câu hỏi --action-id: 
09:43:24 AM 2019-12-11 --User: binh --action: Thêm câu hỏi --action-id: 
09:44:03 AM 2019-12-11 --User: binh --action: Thêm câu hỏi --action-id: 
09:44:46 AM 2019-12-11 --User: binh --action: Thêm câu hỏi --action-id: 
09:45:04 AM 2019-12-11 --User: binh --action: Thêm câu hỏi --action-id: 
09:45:36 AM 2019-12-11 --User: binh --action: Thêm câu hỏi --action-id: 
09:46:15 AM 2019-12-11 --User: binh --action: Thêm câu hỏi --action-id: 
09:46:45 AM 2019-12-11 --User: binh --action: Thêm câu hỏi --action-id: 
09:47:20 AM 2019-12-11 --User: binh --action: Thêm câu hỏi --action-id: 
09:47:54 AM 2019-12-11 --User: binh --action: Thêm câu hỏi --action-id: 
09:48:43 AM 2019-12-11 --User: binh --action: Thêm câu hỏi --action-id: 
09:49:30 AM 2019-12-11 --User: binh --action: Thêm câu hỏi --action-id: 
09:50:11 AM 2019-12-11 --User: binh --action: Thêm câu hỏi --action-id: 
09:50:45 AM 2019-12-11 --User: binh --action: Thêm câu hỏi --action-id: 

src/com/poly/UI/ChonMon.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.poly.UI;

import com.poly.connect.Util;
import com.poly.dao.DeThiDao;
import com.poly.dao.MonHocDao;
import com.poly.dao.TaiKhoanDao;
import com.poly.model.DeThi;
import com.poly.model.TaiKhoan;
import java.io.FileNotFoundException;
import java.io.IOException;
import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.util.ArrayList;
import java.util.logging.Level;
import java.util.logging.Logger;
import javax.swing.JComboBox;
import javax.swing.JOptionPane;

/**
 *
 * @author Tong Duy Hai
 */
public class ChonMon extends javax.swing.JFrame {

    static String monthi;
    static String dethi;
    public static String username;
    public static String ten;
    public static long time;
    public static String maDe;

    /**
     * Creates new form ChonMon
     */
    public ChonMon() {
        try {
            initComponents();
            setLocationRelativeTo(null);
            setResizable(false);
            setUserInfo();
            String monHoc = cbbMonThi.getItemAt(0).toString().trim();
            String maMon = MonHocDao.getMaMonHoc(monHoc);
            cbbDeThi.removeAllItems();
            setSubjectCombox(cbbDeThi, DeThiDao.getListDeThi(maMon));
        } catch (IOException ex) {
            Logger.getLogger(ChonMon.class.getName()).log(Level.SEVERE, null, ex);
        } catch (ClassNotFoundException ex) {
            Logger.getLogger(ChonMon.class.getName()).log(Level.SEVERE, null, ex);
        }
    }

    private void setUserInfo() {
        try {
            username = LoGin.userr;
            TaiKhoan user = TaiKhoanDao.getUser(username);
            mName.setText(user.getUsername().trim());
            mUser.setText(user.getName());
        } catch (Exception e) {
            e.printStackTrace();
        }

    }

    public static void setSubjectCombox(JComboBox comboBox, ArrayList<String> array) {
        for (int i = 0; i < array.size(); i++) {
            comboBox.addItem(array.get(i));
        }
    }

    /**
     * This method is called from within the constructor to initialize the form.
     * WARNING: Do NOT modify this code. The content of this method is always
     * regenerated by the Form Editor.
     */
    @SuppressWarnings("unchecked")
    // <editor-fold defaultstate="collapsed" desc="Generated Code">//GEN-BEGIN:initComponents
    private void initComponents() {

        jMenuItem3 = new javax.swing.JMenuItem();
        jPanel1 = new javax.swing.JPanel();
        jPanel2 = new javax.swing.JPanel();
        cbbDeThi = new javax.swing.JComboBox<>();
        jLabel5 = new javax.swing.JLabel();
        jLabel8 = new javax.swing.JLabel();
        cbbMonThi = new javax.swing.JComboBox<>();
        jPanel3 = new javax.swing.JPanel();
        jLabel2 = new javax.swing.JLabel();
        jLabel3 = new javax.swing.JLabel();
        jLabel4 = new javax.swing.JLabel();
        lblMonthi = new javax.swing.JLabel();
        jLabel6 = new javax.swing.JLabel();
        jLabel7 = new javax.swing.JLabel();
        jButton1 = new javax.swing.JButton();
        jMenuBar1 = new javax.swing.JMenuBar();
        mUser = new javax.swing.JMenu();
        mName = new javax.swing.JMenuItem();
        jMenu1 = new javax.swing.JMenu();
        jSeparator1 = new javax.swing.JPopupMenu.Separator();
        jMenu2 = new javax.swing.JMenu();

        jMenuItem3.setText("jMenuItem3");

        setDefaultCloseOperation(javax.swing.WindowConstants.EXIT_ON_CLOSE);
        setTitle("ÔN THI ĐẠI HỌC KHỐI A");

        cbbDeThi.setFont(new java.awt.Font("Tahoma", 0, 18)); // NOI18N
        cbbDeThi.addItemListener(new java.awt.event.ItemListener() {
            public void itemStateChanged(java.awt.event.ItemEvent evt) {
                cbbDeThiItemStateChanged(evt);
            }
        });
        cbbDeThi.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent evt) {
                cbbDeThiActionPerformed(evt);
            }
        });

        jLabel5.setFont(new java.awt.Font("Tahoma", 0, 18)); // NOI18N
        jLabel5.setText("Chọn đề thi");

        jLabel8.setFont(new java.awt.Font("Tahoma", 0, 18)); // NOI18N
        jLabel8.setText("Chọn môn thi");

        cbbMonThi.setFont(new java.awt.Font("Tahoma", 0, 18)); // NOI18N
        cbbMonThi.setModel(new javax.swing.DefaultComboBoxModel<>(new String[] { "Môn thi", "Toán", "Vật lý", "Hóa học" }));
        cbbMonThi.addItemListener(new java.awt.event.ItemListener() {
            public void itemStateChanged(java.awt.event.ItemEvent evt) {
                cbbMonThiItemStateChanged(evt);
            }
        });
        cbbMonThi.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent evt) {
                cbbMonThiActionPerformed(evt);
            }
        });

        javax.swing.GroupLayout jPanel2Layout = new javax.swing.GroupLayout(jPanel2);
        jPanel2.setLayout(jPanel2Layout);
        jPanel2Layout.setHorizontalGroup(
            jPanel2Layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
            .addGroup(jPanel2Layout.createSequentialGroup()
                .addGap(22, 22, 22)
                .addGroup(jPanel2Layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
                    .addComponent(cbbMonThi, javax.swing.GroupLayout.PREFERRED_SIZE, 268, javax.swing.GroupLayout.PREFERRED_SIZE)
                    .addComponent(jLabel5)
                    .addComponent(jLabel8)
                    .addComponent(cbbDeThi, javax.swing.GroupLayout.PREFERRED_SIZE, 268, javax.swing.GroupLayout.PREFERRED_SIZE))
                .addContainerGap(29, Short.MAX_VALUE))
        );
        jPanel2Layout.setVerticalGroup(
            jPanel2Layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
            .addGroup(jPanel2Layout.createSequentialGroup()
                .addContainerGap()
                .addComponent(jLabel8, javax.swing.GroupLayout.PREFERRED_SIZE, 36, javax.swing.GroupLayout.PREFERRED_SIZE)
                .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.UNRELATED)
                .addComponent(cbbMonThi, javax.swing.GroupLayout.PREFERRED_SIZE, 45, javax.swing.GroupLayout.PREFERRED_SIZE)
                .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.RELATED)
                .addComponent(jLabel5, javax.swing.GroupLayout.PREFERRED_SIZE, 36, javax.swing.GroupLayout.PREFERRED_SIZE)
                .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.RELATED)
                .addComponent(cbbDeThi, javax.swing.GroupLayout.PREFERRED_SIZE, 44, javax.swing.GroupLayout.PREFERRED_SIZE)
                .addContainerGap(javax.swing.GroupLayout.DEFAULT_SIZE, Short.MAX_VALUE))
        );

        jLabel2.setFont(new java.awt.Font("Tahoma", 0, 14)); // NOI18N
        jLabel2.setText("Môn thi");

        jLabel3.setFont(new java.awt.Font("Tahoma", 0, 14)); // NOI18N
        jLabel3.setText("Số câu hỏi");

        jLabel4.setFont(new java.awt.Font("Tahoma", 0, 14)); // NOI18N
        jLabel4.setText("Thời gian");

        lblMonthi.setFont(new java.awt.Font("Tahoma", 0, 14)); // NOI18N

        jLabel6.setFont(new java.awt.Font("Tahoma", 0, 14)); // NOI18N
        jLabel6.setText("20 phút");

        jLabel7.setFont(new java.awt.Font("Tahoma", 0, 14)); // NOI18N
        jLabel7.setText("15 câu");

        jButton1.setBackground(new java.awt.Color(255, 102, 102));
        jButton1.setFont(new java.awt.Font("Tahoma", 0, 18)); // NOI18N
        jButton1.setIcon(new javax.swing.ImageIcon(getClass().getResource("/com/poly/icon/ic_complete.png"))); // NOI18N
        jButton1.setText("START");
        jButton1.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent evt) {
                jButton1ActionPerformed(evt);
            }
        });

        javax.swing.GroupLayout jPanel3Layout = new javax.swing.GroupLayout(jPanel3);
        jPanel3.setLayout(jPanel3Layout);
        jPanel3Layout.setHorizontalGroup(
            jPanel3Layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
            .addGroup(jPanel3Layout.createSequentialGroup()
                .addGap(73, 84, Short.MAX_VALUE)
                .addGroup(jPanel3Layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
                    .addGroup(javax.swing.GroupLayout.Alignment.TRAILING, jPanel3Layout.createSequentialGroup()
                        .addGroup(jPanel3Layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
                            .addGroup(javax.swing.GroupLayout.Alignment.TRAILING, jPanel3Layout.createSequentialGroup()
                                .addComponent(jLabel2, javax.swing.GroupLayout.PREFERRED_SIZE, 71, javax.swing.GroupLayout.PREFERRED_SIZE)
                                .addGap(18, 18, 18)
                                .addComponent(lblMonthi, javax.swing.GroupLayout.PREFERRED_SIZE, 158, javax.swing.GroupLayout.PREFERRED_SIZE))
                            .addGroup(javax.swing.GroupLayout.Alignment.TRAILING, jPanel3Layout.createSequentialGroup()
                                .addComponent(jLabel3, javax.swing.GroupLayout.PREFERRED_SIZE, 71, javax.swing.GroupLayout.PREFERRED_SIZE)
                                .addGap(18, 18, 18)
                                .addComponent(jLabel7, javax.swing.GroupLayout.PREFERRED_SIZE, 158, javax.swing.GroupLayout.PREFERRED_SIZE))
                            .addGroup(javax.swing.GroupLayout.Alignment.TRAILING, jPanel3Layout.createSequentialGroup()
                                .addComponent(jLabel4, javax.swing.GroupLayout.PREFERRED_SIZE, 71, javax.swing.GroupLayout.PREFERRED_SIZE)
                                .addGap(18, 18, 18)
                                .addComponent(jLabel6, javax.swing.GroupLayout.PREFERRED_SIZE, 158, javax.swing.GroupLayout.PREFERRED_SIZE)))
                        .addContainerGap())
                    .addGroup(javax.swing.GroupLayout.Alignment.TRAILING, jPanel3Layout.createSequentialGroup()
                        .addComponent(jButton1, javax.swing.GroupLayout.PREFERRED_SIZE, 240, javax.swing.GroupLayout.PREFERRED_SIZE)
                        .addGap(43, 43, 43))))
        );
        jPanel3Layout.setVerticalGroup(
            jPanel3Layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
            .addGroup(jPanel3Layout.createSequentialGroup()
                .addContainerGap()
                .addGroup(jPanel3Layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
                    .addComponent(lblMonthi, javax.swing.GroupLayout.PREFERRED_SIZE, 32, javax.swing.GroupLayout.PREFERRED_SIZE)
                    .addComponent(jLabel2, javax.swing.GroupLayout.PREFERRED_SIZE, 32, javax.swing.GroupLayout.PREFERRED_SIZE))
                .addGap(18, 18, 18)
                .addGroup(jPanel3Layout.createParallelGroup(javax.swing.GroupLayout.Alignment.BASELINE)
                    .addComponent(jLabel6, javax.swing.GroupLayout.PREFERRED_SIZE, 32, javax.swing.GroupLayout.PREFERRED_SIZE)
                    .addComponent(jLabel4, javax.swing.GroupLayout.PREFERRED_SIZE, 32, javax.swing.GroupLayout.PREFERRED_SIZE))
                .addGap(18, 18, 18)
                .addGroup(jPanel3Layout.createParallelGroup(javax.swing.GroupLayout.Alignment.BASELINE)
                    .addComponent(jLabel7, javax.swing.GroupLayout.PREFERRED_SIZE, 32, javax.swing.GroupLayout.PREFERRED_SIZE)
                    .addComponent(jLabel3, javax.swing.GroupLayout.PREFERRED_SIZE, 32, javax.swing.GroupLayout.PREFERRED_SIZE))
                .addGap(18, 18, 18)
                .addComponent(jButton1, javax.swing.GroupLayout.PREFERRED_SIZE, 59, javax.swing.GroupLayout.PREFERRED_SIZE)
                .addContainerGap(24, Short.MAX_VALUE))
        );

        javax.swing.GroupLayout jPanel1Layout = new javax.swing.GroupLayout(jPanel1);
        jPanel1.setLayout(jPanel1Layout);
        jPanel1Layout.setHorizontalGroup(
            jPanel1Layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
            .addGroup(jPanel1Layout.createSequentialGroup()
                .addComponent(jPanel2, javax.swing.GroupLayout.PREFERRED_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.PREFERRED_SIZE)
                .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.RELATED)
                .addComponent(jPanel3, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, Short.MAX_VALUE))
        );
        jPanel1Layout.setVerticalGroup(
            jPanel1Layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
            .addGroup(jPanel1Layout.createSequentialGroup()
                .addGap(0, 8, Short.MAX_VALUE)
                .addGroup(jPanel1Layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING, false)
                    .addComponent(jPanel2, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, Short.MAX_VALUE)
                    .addComponent(jPanel3, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, Short.MAX_VALUE)))
        );

        mUser.setIcon(new javax.swing.ImageIcon(getClass().getResource("/com/poly/icon/ic_user_1.png"))); // NOI18N
        mUser.setText("Ten");

        mName.setText("Tong Duy Hai");
        mUser.add(mName);

        jMenuBar1.add(mUser);

        jMenu1.setIcon(new javax.swing.ImageIcon(getClass().getResource("/com/poly/icon/ic_logout.png"))); // NOI18N
        jMenu1.setText("Đăng xuất");
        jMenu1.addMouseListener(new java.awt.event.MouseAdapter() {
            public void mouseClicked(java.awt.event.MouseEvent evt) {
                jMenu1MouseClicked(evt);
            }
        });
        jMenu1.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent evt) {
                jMenu1ActionPerformed(evt);
            }
        });
        jMenu1.add(jSeparator1);

        jMenuBar1.add(jMenu1);

        jMenu2.setIcon(new javax.swing.ImageIcon(getClass().getResource("/com/poly/icon/ic_stop.png"))); // NOI18N
        jMenu2.setText("Kết thúc");
        jMenu2.addMouseListener(new java.awt.event.MouseAdapter() {
            public void mouseClicked(java.awt.event.MouseEvent evt) {
                jMenu2MouseClicked(evt);
            }
        });
        jMenuBar1.add(jMenu2);

        setJMenuBar(jMenuBar1);

        javax.swing.GroupLayout layout = new javax.swing.GroupLayout(getContentPane());
        getContentPane().setLayout(layout);
        layout.setHorizontalGroup(
            layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
            .addComponent(jPanel1, javax.swing.GroupLayout.Alignment.TRAILING, javax.swing.GroupLayout.PREFERRED_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.PREFERRED_SIZE)
        );
        layout.setVerticalGroup(
            layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
            .addGroup(layout.createSequentialGroup()
                .addComponent(jPanel1, javax.swing.GroupLayout.PREFERRED_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.PREFERRED_SIZE)
                .addGap(0, 0, Short.MAX_VALUE))
        );

        pack();
    }// </editor-fold>//GEN-END:initComponents

    private void cbbDeThiActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_cbbDeThiActionPerformed
        // TODO add your handling code here:

    }//GEN-LAST:event_cbbDeThiActionPerformed

    private void jButton1ActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_jButton1ActionPerformed
        // TODO add your handling code here:
        if (cbbDeThi.getSelectedItem() == null) {
            JOptionPane.showMessageDialog(this, "Vui lòng chọn đề thi");
        } else {
            String maDe = cbbDeThi.getSelectedItem().toString();
            Util.changeLog(username, "Làm bài thi", maDe);
            VaoThi hsTest = new VaoThi();
            dispose();
            hsTest.setVisible(true);
            this.setVisible(false);
        }
    }//GEN-LAST:event_jButton1ActionPerformed

    private void cbbMonThiActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_cbbMonThiActionPerformed
        // TODO add your handling code here:
        monthi = cbbMonThi.getSelectedItem().toString().trim();
        lblMonthi.setText(monthi);
    }//GEN-LAST:event_cbbMonThiActionPerformed

    private void jMenu1ActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_jMenu1ActionPerformed
        // TODO add your handling code here:

    }//GEN-LAST:event_jMenu1ActionPerformed

    private void jMenu1MouseClicked(java.awt.event.MouseEvent evt) {//GEN-FIRST:event_jMenu1MouseClicked
        // TODO add your handling code here:
        int chon = JOptionPane.showConfirmDialog(this, "Bạn muốn đăng xuất ?", "Đăng xuất", JOptionPane.YES_NO_OPTION);
        if (chon == JOptionPane.YES_OPTION) {
            this.dispose();
            this.setVisible(false);
            LoGin logout = new LoGin();
            logout.setLocationRelativeTo(null);
            logout.setVisible(true);
        }
    }//GEN-LAST:event_jMenu1MouseClicked

    private void jMenu2MouseClicked(java.awt.event.MouseEvent evt) {//GEN-FIRST:event_jMenu2MouseClicked
        // TODO add your handling code here:
        int chon = JOptionPane.showConfirmDialog(this, "Bạn muốn kết thúc ?", "KT", JOptionPane.YES_NO_OPTION);
        if (chon == JOptionPane.YES_OPTION) {
            System.exit(0);
        }
    }//GEN-LAST:event_jMenu2MouseClicked

    private void cbbMonThiItemStateChanged(java.awt.event.ItemEvent evt) {//GEN-FIRST:event_cbbMonThiItemStateChanged
        // TODO add your handling code here:
        try {
            String monHoc = cbbMonThi.getSelectedItem().toString().trim();
            String maMon = MonHocDao.getMaMonHoc(monHoc);
            cbbDeThi.removeAllItems();
            setSubjectCombox(cbbDeThi, DeThiDao.getListDeThi(maMon));
        } catch (Exception e) {
            e.printStackTrace();
        }

    }//GEN-LAST:event_cbbMonThiItemStateChanged

    private void cbbDeThiItemStateChanged(java.awt.event.ItemEvent evt) {//GEN-FIRST:event_cbbDeThiItemStateChanged
        // TODO add your handling code here:
        try {
            String monHoc = cbbMonThi.getSelectedItem().toString().trim();
            String maMon = MonHocDao.getMaMonHoc(monHoc);
            if (cbbDeThi.getSelectedItem() != null) {
                maDe = cbbDeThi.getSelectedItem().toString().trim();
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
    }//GEN-LAST:event_cbbDeThiItemStateChanged

    /**
     * @param args the command line arguments
     */
    public static void main(String args[]) {
        /* Set the Nimbus look and feel */
        //<editor-fold defaultstate="collapsed" desc=" Look and feel setting code (optional) ">
        /* If Nimbus (introduced in Java SE 6) is not available, stay with the default look and feel.
         * For details see http://download.oracle.com/javase/tutorial/uiswing/lookandfeel/plaf.html 
         */
        try {
            for (javax.swing.UIManager.LookAndFeelInfo info : javax.swing.UIManager.getInstalledLookAndFeels()) {
                if ("Nimbus".equals(info.getName())) {
                    javax.swing.UIManager.setLookAndFeel(info.getClassName());
                    break;
                }
            }
        } catch (ClassNotFoundException ex) {
            java.util.logging.Logger.getLogger(ChonMon.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        } catch (InstantiationException ex) {
            java.util.logging.Logger.getLogger(ChonMon.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        } catch (IllegalAccessException ex) {
            java.util.logging.Logger.getLogger(ChonMon.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        } catch (javax.swing.UnsupportedLookAndFeelException ex) {
            java.util.logging.Logger.getLogger(ChonMon.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        }
        //</editor-fold>

        /* Create and display the form */
        java.awt.EventQueue.invokeLater(new Runnable() {
            public void run() {
                new ChonMon().setVisible(true);
            }
        });
    }

    // Variables declaration - do not modify//GEN-BEGIN:variables
    private javax.swing.JComboBox<String> cbbDeThi;
    private javax.swing.JComboBox<String> cbbMonThi;
    private javax.swing.JButton jButton1;
    private javax.swing.JLabel jLabel2;
    private javax.swing.JLabel jLabel3;
    private javax.swing.JLabel jLabel4;
    private javax.swing.JLabel jLabel5;
    private javax.swing.JLabel jLabel6;
    private javax.swing.JLabel jLabel7;
    private javax.swing.JLabel jLabel8;
    private javax.swing.JMenu jMenu1;
    private javax.swing.JMenu jMenu2;
    private javax.swing.JMenuBar jMenuBar1;
    private javax.swing.JMenuItem jMenuItem3;
    private javax.swing.JPanel jPanel1;
    private javax.swing.JPanel jPanel2;
    private javax.swing.JPanel jPanel3;
    private javax.swing.JPopupMenu.Separator jSeparator1;
    private javax.swing.JLabel lblMonthi;
    private javax.swing.JMenuItem mName;
    private javax.swing.JMenu mUser;
    // End of variables declaration//GEN-END:variables
}
